pandas/tests/series/methods/test_fillna.py

```diff
@@ -671,6 +671,28 @@ def test_fillna_categorical_with_new_categories(self, fill_value, expected_outpu
         result = ser.fillna(fill_value)
         tm.assert_series_equal(result, exp)
 
+    @pytest.mark.parametrize(
+        "fill_value, expected_output",
+        [
+            ("B", ["A", "B", "B", "B", "C"]),
+            ("C", ["A", "B", "C", "C", "C"])
+        ],
+    )
+
+    def test_series_fill(fill_value, expected_output):
+        # GH#32414
+        data = ["A", "B", np.nan, np.nan, "C"]
+        ser = Series(Categorical(data, categories=["A", "B", "C"]))
+
+        # msg = "Element not present in categories. Cannot be filled in series."
+        # with pytest.raises(TypeError, match=msg):
+        #     ser.fillna("D")
+
+        exp = Series(Categorical(expected_output, categories=["A", "B", "C"]))
+        result = ser.fillna(fill_value)
```

Comment:
This does not cover the issue. We need to check with a categorical fill_value too.

Reply:
I have defined the the fill_value in the pytest parameterized section. There I have provided a fill_value . Do we need to explicitly define that categroical value here??

Reply:
Could you please ellaborate on this a little more ?? I find it difficult to understand

Reply:
You are not testing everything mentioned in the issue. @mroeschke provided a code snippet there which should be covered here

Reply:
In [22]: cat = pd.Categorical(["A", "B", None, "A"])
    ...: ser = pd.Series(cat).fillna("B")

In [23]: >>> filled = cat.fillna(ser)

In [24]: >>> cat.fillna(filled)
Out[24]:
['A', 'B', 'B', 'A']
Categories (2, object): ['A', 'B']

Reply:
hey @phofl, this was the code snippet as being provided by @mroeschke, I have tried to add the same thing. Soo i'm asking should I add "B" instead of that fill_value?? 
I saw other tests too and they did the same thing, do you want me to explicitly declare "B"

Reply:
okay @phofl , I have checked with categorical_fillna too. I will add the updated code tomorrow positively. I think with that issue will be resolved. Thanks